app/controllers/catalog_controller.rb

```diff
@@ -1142,25 +1162,33 @@ def trees_to_replace(trees)
     trees_to_replace
   end
 
+  def entry_point_keys
+    {:fqname => 'Provision', :reconfigure_fqname => 'Reconfigure', :retire_fqname => 'Retirement'}.map do |item|
+      {:edit_key => item [0], :name => item[1], :configuration_script_id => params["#{item}_configuration_script_id"]}
+    end
+  end
+
   def set_resource_action(st)
+    puts "\n\n\nst = #{st.inspect}"
+    puts "\n\n\@edit[:new] = #{@edit[:new].inspect}"
+    puts "\n\n\@params = #{params.inspect}"
+    byebug
     d = @edit[:new][:dialog_id].nil? ? nil : Dialog.where(:id => @edit[:new][:dialog_id]).first
-    actions = [
-      {:name => 'Provision', :edit_key => :fqname},
-      {:name => 'Reconfigure', :edit_key => :reconfigure_fqname},
-      {:name => 'Retirement', :edit_key => :retire_fqname}
-    ]
-    actions.each do |action|
-      ra = st.resource_actions.find_by(:action => action[:name])
-      if ra.nil? && @edit[:new][action[:edit_key]].present?
-        attrs = {:action        => action[:name],
-                 :ae_attributes => {:service_action => action[:name]}}
-        ra = st.resource_actions.build(attrs)
-      end
+    
+    entry_point_keys.each do |action|
+      ra = st.resource_actions.create_with(
+        :action                  => action[:name],
+        :configuration_script_id => action[:configuration_script_id],
+        :ae_attributes           => {:service_action => action[:name]}
+      ).find_or_create_by!(:action => action[:name])
+
+
       if @edit[:new][action[:edit_key]].blank?
         st.resource_actions.where(:action => action[:name]).first.try(:destroy)
       else
-        ra.update(:dialog => d, :fqname => @edit[:new][action[:edit_key]])
+        ra.update(:dialog => d, :fqname => @edit[:new][action[:name]])
```

Comment:
This will set `fqname` to `Provision` which isn't right

Reply:
NOTE I noticed that it wasn't possible to clear the ae_* attributes by doing `ra.update(:fqname => nil)`, opened a PR to do that https://github.com/ManageIQ/manageiq/pull/22600